make cult play along with servo

p1-0tr opened the pull request:
Hi,

I stumbled on this issue - https://github.com/servo/servo/issues/6710 - and as it sparked my interest, I started playing around with making servo produce noises (which might be a bit insane as I am only starting my rust adventure, but nonetheless I'd like to try). I've rebased dmarcos' change mentioned in the issue and fixed-up some problems. In the process I also had to make a couple changes to cult, which is why I am making this pull request. I tested my changes on OSX, I also did a successful android build (but I don't have a device recent enough to run servo on).

I hope my changes make sense :)

Greetings,
Piotr

p1-0tr committed:
change test to use normal mutex and condvar

Signed-off-by: Piotr Stankiewicz <bionicrift@gmail.com>

p1-0tr committed:
make cross-builds easier (mainly - link against cubeb statically)

Libcubeb isn't available out-of-the-box on any platform, so linking
against it statically will make any binary using cult more portable.
Also, with the addition of the option to build libcubeb('build-cubeb'
cargo feature) along cult, it makes cross-builds far easier.

Beside that two minor issues that prevented builds on a non x64 machine
were fixed.

Signed-off-by: Piotr Stankiewicz <bionicrift@gmail.com>

p1-0tr committed:
dummy heapsize support (for use in servo)

Signed-off-by: Piotr Stankiewicz <bionicrift@gmail.com>

p1-0tr commented:
Hi,

I decided curl'ing cubeb into the build felt like too much of a hacky solution, so I made it a submodule (I am not the hugest fan of those, but I think they make sense for managing optional dependencies). 

Greetings,
Piotr

padenot commented:
Thanks!

Can we use cubeb's build system instead of a quick reimplementation in Rust ? That will make it easier to keep up with upstream (that is moving a lot at the moment, with upcoming support for input, i.e. microphone). It's using libtool, so we can probably just ask libtool to create a static library and link with that.

p1-0tr commented:
Sure, I already have some code to do that :) I went with the "makefile.cargo" solution to prevent from depending on having autotools.

p1-0tr committed:
improve how cubeb is linked

Instead of always linking cubeb statically, use dynamic linking if
build-cubeb is not set. Also, when build-cubeb is set, figure out what
middle-ware should be linked based on cubeb's config.h.

Signed-off-by: Piotr Stankiewicz <bionicrift@gmail.com>
